packages/metro/src/ModuleGraph/worker/__tests__/collectDependencies-test.js

```diff
@@ -32,94 +32,24 @@ const nullthrows = require('nullthrows');
 
 const {any, objectContaining} = expect;
 
-const {InvalidRequireCallError, DefaultModuleDependencyRegistry} =
-  collectDependencies;
+const {InvalidRequireCallError} = collectDependencies;
 const opts = {
   asyncRequireModulePath: 'asyncRequire',
   dynamicRequires: 'reject',
   inlineableCalls: [],
   keepRequireNames: true,
   allowOptionalDependencies: false,
   dependencyMapName: null,
+  unstable_allowRequireContext: true,
 };
 
-describe('DefaultModuleDependencyRegistry', () => {
-  describe('getKeyForDependency', () => {
-    it(`generates basic key`, () => {
-      const registry = new DefaultModuleDependencyRegistry();
-      expect(
-        registry.getKeyForDependency({
-          name: './file.js',
-          optional: false,
-          asyncType: null,
-        }),
-      ).toBe('./file.js');
-    });
-    it(`generates require.context key`, () => {
-      const registry = new DefaultModuleDependencyRegistry();
-
-      expect(
-        registry.getKeyForDependency({
-          name: './dir',
-          optional: false,
-          asyncType: null,
-          contextParams: {
-            recursive: true,
-            filter: /foobar/,
-            mode: 'lazy',
-          },
-        }),
-      ).toBe('./dir__context__true__/foobar/__lazy');
-    });
-  });
-
-  it(`registers dependencies without collision`, () => {
-    const registry = new DefaultModuleDependencyRegistry();
-
-    registry.registerDependency({
-      name: './dir',
-      optional: false,
-      asyncType: null,
-    });
-
-    registry.registerDependency({
-      name: './dir',
-      optional: false,
-      asyncType: null,
-      contextParams: {
-        recursive: true,
-        filter: /foobar/,
-        mode: 'lazy',
-      },
-    });
-
-    expect(registry.getDependencies()).toEqual([
-      {
-        asyncType: undefined,
-        index: 0,
-        locs: [],
-        name: './dir',
-      },
-      {
-        asyncType: undefined,
-        contextParams: {
-          filter: /foobar/,
-          mode: 'lazy',
-          recursive: true,
-        },
-        index: 1,
-        locs: [],
-        name: './dir',
-      },
-    ]);
-  });
-});
-
 describe(`require.context`, () => {
   it(`collects require context arguments`, () => {
     const ast = astFromCode(`
   const a = require.context('./', false, /foobar/m, 'eager');
   const b = require.context('./baz');
+  const c = require.context('./dir', true);
+  const d = require.context('./dir');
```

Comment:
Needs a few more test cases, e.g. what happens when we have the same directory etc but different values for `recursive`, plus everything else that can cause the dependency key to be different. (This will be a lot, so ideally these would be separate test cases with clear descriptions as opposed to just additional lines in this one test case.)

Reply:
imo we had this before https://github.com/facebook/metro/pull/821/commits/7c19f6e244611b114b6aa72e1ef996ac4e9f5acd not sure how to proceed short of recreating those tests.

Reply:
Those tests were in terms of `DefaultModuleDependencyRegistry`. I'm thinking instead we should clone the whole `collects require context arguments` test multiple times, so that each test uses `collectDependencies` and covers a particular scenario/edge case in isolation:

1. contexts differ only by directory
2. contexts differ only by `recursive`
3. contexts differ only by regex pattern
4. contexts differ only by regex flags
5. contexts differ only by `mode`
6. default values in case of `undefined` or missing arguments
7. multiple contexts that are identical
8. contexts differ only by trailing slash in directory (e.g `./dir` vs `./dir/`)
9. ...

Reply:
Also, importantly, a test for `require('./')` and `require.context('./')` being distinct.